### PR TITLE
feat(upload-sinan-dbf): include trigger to uploading files into dbf2/uploaded/

### LINKED ISSUE #16: Implement automation for SINAN data upload and processing in UpdateHistoricoAlerta
#### Objective:
To automate the process of uploading and processing SINAN data files (.dbf) within the UpdateHistoricoAlerta repository, ensuring a streamlined and error-free operation that integrates seamlessly with our existing Ansible and Python best practices.

#### Requirements:
1. **New Makim target for SINAN data processing**:
    - Implement a new target within the `.makim.yaml` to handle the upload and insertion of SINAN data files. This target should accept arguments necessary for the CLI parameters of the `load_sinan` Django management command.
    - Example execution command: 
    ```bash
    python AlertaDengue/manage.py load_sinan /Storage/infodengue_data/sftp2/alertadengue/BR-DEN-2024-02-26-2024.dbf 2024 --default-cid A90
    ```
    - **Arguments**:
        - File Name (assumed to be in a default path to simplify argument passing).
        - Year.
        - CID (default A90 if not specified).

2. **Ansible Playbook for File Upload and Preparation**:
    - Develop an Ansible playbook dedicated to preparing the environment and executing the `load_sinan` command. The playbook will encompass the following steps:
        1. **Upload DBF to Server**: Automate the transfer of the `.dbf` file into the designated directory (`/Storage/infodengue_data/sftp2/alertadengue/uploaded/`) on the HETZNER server.
        2. **Change Working Directory**: Shift to the AlertaDengue repository directory where the Django command will be executed.
        3. **Activate Conda Environment**: Ensure the `dev-alertadengue` Conda environment is activated to meet the execution dependencies.
        4. **Execute `load_sinan` Command**: Run the `load_sinan` command with the provided CLI arguments to process and insert the data.
        5. **Cleanup**: Remove the uploaded `.dbf` file from the local machine after successful data insertion.
        6. **Logging**: Maintain a log of the operation in the tmp/ directory for tracking and auditing purposes.

#### Implementation Notes:
- Refer to existing playbooks within the UpdateHistoricoAlerta repository for consistent playbook structure and best practices.

### PR DESCRIPTION
**Pull request by luabida**
Closes #16.

### CODE REVIEW

File: playbooks/upload-sinan-dbf.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- name: Upload SINAN DBF into Production
+  hosts: cluster
+  gather_facts: true
+
+  vars:
+    HOST_FILE_NAME: "{{ file }}"
+    YEAR: "{{ year }}"
+    CID: "{{ cid }}"
+    DBF_HOST_DIR: ../ansible/roles/sinan/dbf
+    HOST_FILE_PATH: "{{ DBF_HOST_DIR }}/{{ HOST_FILE_NAME }}"
```
Comment (esloch) [inline playbooks/upload-sinan-dbf.yaml:11]:
```python
"{{ DBF_HOST_DIR }}/{{ DBF_NAME }}.dbf"
or
"{{ HOST_DIR }}/{{ DBF_NAME }}.dbf"
```

File: playbooks/upload-sinan-dbf.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        port: "{{ cluster_node_port }}"
+        host: "{{ ansible_host }}"
+        timeout: 10
+      ignore_errors: true
+      register: desired_port_check
+
+    - name: Copying DBF to Production
+      copy:
+        src: "{{ HOST_FILE_PATH }}"
+        dest: "{{ REMOTE_TARGET_DIR }}/{{ HOST_FILE_NAME }}"
```
Comment (esloch) [inline playbooks/upload-sinan-dbf.yaml:28]:
```python
 dest: "{{ REMOTE_TARGET_DIR }}/{{ DBF_NAME }}.dbf"
 ```

**Review by esloch — CHANGES_REQUESTED** [review]
Could you provide more information on how the command line is executed, for example?

**Comment by luabida** [comment]
> Could you provide more information on how the command line is executed, for example?

before the changes the commands were:
`makim ansible.upload-sinan-dbf --file test.txt --year 2024`
`makim ansible.history --logfile sinan_upload_dbf.log`

**Comment by luabida** [comment]
@esloch 

command:
`makim ansible.upload-sinan-dbf --file-path /tmp/test.dbf --year 2024`

output:
`Tue Mar 12 04:10:30 PM -03 2024 ==> BR-DENG-2024-03-12-2024.dbf uploaded successfully`

**Comment by luabida** [comment]
command:
`makim ansible.upload-sinan-dbf --file-path /tmp/test.dbf --year 2000 --codarea "SC" --disease "ZIKA"`

output:
`Tue Mar 12 04:20:02 PM -03 2024 ==> SC-ZIKA-2000-03-12-2024.dbf uploaded successfully`

**Comment by luabida** [comment]
OBS: `file-path` can't be a relative path, it must be an absolute path

File: playbooks/upload-sinan-dbf.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      ignore_errors: true
+      register: desired_port_check
+
+    - name: Copying file to Production
+      copy:
+        src: "{{ HOST_FILE_PATH }}"
+        dest: "{{ REMOTE_FILE_PATH }}"
+        owner: "{{ cluster_user_name }}"
+        group: "{{ cluster_user_name }}"
+
```
Comment (esloch) [inline playbooks/upload-sinan-dbf.yaml:35]: Please add the task to run the Django command `load_sinan`.

**Review by esloch — CHANGES_REQUESTED** [review]

File: playbooks/upload-sinan-dbf.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    - name: Executing {{ SCRIPT }} on remote server
+      become: true
+      become_user: administrador
+      shell: "nohup /tmp/{{ SCRIPT }} /MEDIA_ROOT/{{ FILE_NAME }} {{ NOTIF_YEAR }} {{ CID }}"
```
Comment (esloch) [inline playbooks/upload-sinan-dbf.yaml:51]: If I'm not mistaken, the /MEDIA_ROOT/ is a volume in the web AlertaDengue container, not a directory in the root filesystem..

**Comment by luabida** [comment]
ready to merge

**Review by esloch — APPROVED** [review]
Thank you, Bida, for working on that implementation. Could you proceed with the merge?

**Pull request merged**